xds/internal/xdsclient/xdslbregistry/test/converter_test.go

```diff
@@ -0,0 +1,381 @@
+				Name: "round_robin",
+			},
+			rhDisabled: true,
+		},
+		{
+			name: "custom_lb_type_v3_struct",
+			policy: &v3clusterpb.LoadBalancingPolicy{
+				Policies: []*v3clusterpb.LoadBalancingPolicy_Policy{
+					{
+						TypedExtensionConfig: &v3corepb.TypedExtensionConfig{
+							TypedConfig: testutils.MarshalAny(&v3cncftypepb.TypedStruct{
+								TypeUrl: "type.googleapis.com/myorg.MyCustomLeastRequestPolicy",
+								Value:   &structpb.Struct{},
+							}),
+						},
+					},
+				},
+			},
+			wantConfig: &internalserviceconfig.BalancerConfig{
+				Name:   "myorg.MyCustomLeastRequestPolicy",
+				Config: customLBConfig{},
+			},
+		},
+		{
+			name: "custom_lb_type_v1_struct",
+			policy: &v3clusterpb.LoadBalancingPolicy{
+				Policies: []*v3clusterpb.LoadBalancingPolicy_Policy{
+					{
+						TypedExtensionConfig: &v3corepb.TypedExtensionConfig{
+							TypedConfig: testutils.MarshalAny(&v1udpatypepb.TypedStruct{
+								TypeUrl: "type.googleapis.com/myorg.MyCustomLeastRequestPolicy",
+								Value:   &structpb.Struct{},
+							}),
+						},
+					},
+				},
+			},
+			wantConfig: &internalserviceconfig.BalancerConfig{
+				Name:   "myorg.MyCustomLeastRequestPolicy",
+				Config: customLBConfig{},
+			},
+		},
+		{
+			name: "wrr_locality_child_round_robin",
+			policy: &v3clusterpb.LoadBalancingPolicy{
+				Policies: []*v3clusterpb.LoadBalancingPolicy_Policy{
+					{
+						TypedExtensionConfig: &v3corepb.TypedExtensionConfig{
+							TypedConfig: wrrLocalityAny(&v3roundrobinpb.RoundRobin{}),
+						},
+					},
+				},
+			},
+			wantConfig: &internalserviceconfig.BalancerConfig{
+				Name: wrrlocality.Name,
+				Config: &wrrlocality.LBConfig{
+					ChildPolicy: &internalserviceconfig.BalancerConfig{
+						Name: "round_robin",
+					},
+				},
+			},
+		},
+		{
+			name: "wrr_locality_child_custom_lb_type_v3_struct",
+			policy: &v3clusterpb.LoadBalancingPolicy{
+				Policies: []*v3clusterpb.LoadBalancingPolicy_Policy{
+					{
+						TypedExtensionConfig: &v3corepb.TypedExtensionConfig{
+							TypedConfig: wrrLocalityAny(&v3cncftypepb.TypedStruct{
+								TypeUrl: "type.googleapis.com/myorg.MyCustomLeastRequestPolicy",
+								Value:   &structpb.Struct{},
+							}),
+						},
+					},
+				},
+			},
+			wantConfig: &internalserviceconfig.BalancerConfig{
+				Name: wrrlocality.Name,
+				Config: &wrrlocality.LBConfig{
+					ChildPolicy: &internalserviceconfig.BalancerConfig{
+						Name:   "myorg.MyCustomLeastRequestPolicy",
+						Config: customLBConfig{},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.rhDisabled {
+				oldRingHashSupport := envconfig.XDSRingHash
+				envconfig.XDSRingHash = false
+				defer func() {
+					envconfig.XDSRingHash = oldRingHashSupport
+				}()
+			}
+			rawJSON, err := xdslbregistry.ConvertToServiceConfig(test.policy)
+			if err != nil {
+				t.Fatalf("unwanted error in ConvertToServiceConfig: %v", err)
```

Comment:
Prefer writing this error string as follows:
`t.Fatalf("ConvertToServiceConfig(%s) failed: %v", pretty.ToJSON(test.policy), err)`

See: go/go-style/decisions#identify-the-input

Reply:
Done. Thanks for the link/suggestion.